Source/VMware.vSphereDSC/Classes/InventoryBaseDSC.ps1

```diff
@@ -0,0 +1,92 @@
+#>
+
+class InventoryBaseDSC : BaseDSC {
+    <#
+    .DESCRIPTION
+
+    The full path to the Datacenter we will use from the Inventory.
+    The parts of the path are separated with "/" where the last part of the path is the Datacenter name.
+    #>
+    [DscProperty(Key)]
+    [string] $DatacenterPath
+
+    <#
+    .DESCRIPTION
+
+    Value indicating if the Inventory object should be Present or Absent.
+    #>
+    [DscProperty(Mandatory)]
+    [Ensure] $Ensure
+
+    <#
+    .DESCRIPTION
+
+    Returns the Datacenter we will use from the Inventory.
+    #>
+    [PSObject] GetDatacenterFromPath() {
+        if ($this.DatacenterPath -eq [string]::Empty) {
+            throw "You have passed an empty path which is not a valid value."
+        }
+
+        $vCenter = $this.Connection
+        $rootFolder = Get-View -Server $this.Connection $vCenter.ExtensionData.Content.RootFolder
+
+        <#
+        This is a special case where only the Datacenter name is passed.
+        So we check if there is a Datacenter with that name at root folder.
+        #>
+        if ($this.DatacenterPath -NotContains '/') {
+            try {
+                $datacenter = Get-Datacenter -Server $this.Connection -Name $this.DatacenterPath -Location $rootFolder -ErrorAction Stop
+                return $datacenter
+            }
+            catch {
+                throw "Datacenter with name $($this.DatacenterPath) was not found at $rootFolder. For more inforamtion: $($_.Exception.Message)"
+            }
+        }
+
+        $pathItems = $this.DatacenterPath -Split '/'
+        $datacenterName = $pathItems[$pathItems.Length - 1]
+
+        # Removes the Datacenter name from the path items array as we already retrieved it.
+        $pathItems = $pathItems[0..($pathItems.Length - 2)]
+
+        $childEntities = Get-View -Server $this.Connection $rootFolder.ChildEntity
+        $foundPathItem = $null
+
+        foreach ($pathItem in $pathItems) {
+            $foundPathItem = $childEntities | Where-Object -Property Name -eq $pathItem
+
+            if ($null -eq $foundPathItem) {
+                throw "$pathItem could not be found in the Inventory. Please verify you have passed a valid path."
+            }
+
+            # If the found path item is Folder and not Datacenter we start looking in the items of this Folder.
+            if ($foundPathItem.GetType().Name -eq 'Folder') {
```

Comment:
Better use full type name and -is operator. In case inheritor of Folder is introduced this if will fail.

$foundPathItem could be an array of Folder and Datacenter in case there is Folder and Datacenter with the same name on the same tree layer. Handle this case as well